File: C version/array/moveZeroes.c
#include<stdio.h>
void moveZeroes(int* nums, int numsSize){
    int i = 0;
    int n = 0;
    for(;i<numsSize;i++){
        if(nums[i]!=0){
            nums[n] = nums[i];
            n++;
        }
    }
    for(;n<numsSize;n++){
        nums[n] = 0;
    }
    for(i=0;i<numsSize;i++){
        printf("%d ",nums[i]);
    }
}
void main(){
    int a[5]={0,1,0,3,12};
    moveZeroes(a,5);
    getchar();
}

File: Rust-version/src/problem/move_zeroes.rs
use super::Solution;

impl Solution {
    pub fn move_zeroes(nums: &mut Vec<i32>) {
        let mut cur_index = 0;
        for i in 0..nums.len() {
            if nums[i]!=0 {
                nums[cur_index]=nums[i];
                cur_index+=1;
            }
        }
        while cur_index < nums.len() {
            nums[cur_index] = 0;
            cur_index+=1;
        }
        print!("{:?}",nums);
    }
}